opennlp-tools/src/main/java/opennlp/tools/sentdetect/segment/SentenceTokenizerME.java

```diff
@@ -0,0 +1,235 @@
+    }
+    if (end < text.length()) {
+      end = text.length();
+      sentence = text.subSequence(start, end).toString();
+      sentence = removeSpace(sentence);
+      if (sentence != null) {
+        sentenceList.add(sentence);
+      }
+    }
+    return sentenceList;
+  }
+
+  public String removeSpace(String segment) {
+    if (segment != null) {
+      int first = 0;
+      int last = segment.length();
+      while (first < segment.length() && StringUtil.isWhitespace(segment.charAt(first))) {
+        first++;
+      }
+      while (last > 0 && StringUtil.isWhitespace(segment.charAt(last - 1))) {
+        last--;
+      }
+      if (last - first > 0) {
+        return segment.substring(first, last);
+      }
+    }
+    return null;
+  }
+
+  public Set<Integer> getBreakSections() {
+    if (breakSections == null) {
+      breakSections = new TreeSet<Integer>();
+      for (Rule rule : languageTool.getBreakRuleList()) {
+
+        Pattern beforePattern = languageTool.compile(rule.getBeforePattern());
+        Pattern afterPattern = languageTool.compile(rule.getAfterPattern());
+        this.beforeMatcher = beforePattern.matcher(text);
+        this.afterMatcher = afterPattern.matcher(text);
+        this.found = true;
+        while (find()) {
+          breakSections.add(getBreakPosition());
+        }
+      }
+    }
+    return breakSections;
+  }
+
+  private boolean find() {
+    found = false;
+    while ((!found) && beforeMatcher.find()) {
+      afterMatcher.region(beforeMatcher.end(), text.length());
+      found = afterMatcher.lookingAt();
+    }
+    return found;
+  }
+
+  private int getBreakPosition() {
+    return afterMatcher.start();
+  }
+
+  public List<Section> getNoBreakSections() {
+    if (noBreakSections == null) {
+      noBreakSections = new ArrayList<Section>();
+      Pattern pattern = languageTool.getNoBreakPattern();
+      Matcher matcher = pattern.matcher(getText());
+      while (matcher.find()) {
+        noBreakSections.add(new Section(matcher.start(), matcher.end()));
+      }
+    }
+    return noBreakSections;
+  }
+
+  public CharSequence getText() {
+    if (text == null) {
+      text = read(bufferLength + 1);
+    }
+    return text;
+  }
+
+  private String read(int amount) {
+    char[] charBuffer = new char[amount];
+    int count = read(reader, charBuffer);
+
+    String result;
+    if (count == amount) {
+      result = new String(charBuffer, 0, count - 1);
+    } else if (count > 0 && count < amount) {
+      result = new String(charBuffer, 0, count);
+    } else {
+      result = "";
+    }
+
+    return result;
+  }
+
+  private int read(Reader reader, char[] buffer) {
+
+    int start = 0;
+    int count;
+
+    try {
+      while (true) {
+        if (!(((count = reader.read(buffer, start, buffer.length - start)) != -1)
+            && start < buffer.length)) {
+          break;
+        }
+        start += count;
+      }
+    } catch (IOException e) {
+      e.printStackTrace();
```

Comment:
We shouldn*t just print the stack trace: Either rethrow as runtime exception or at least log it.